Airlock painter now printable in autolathe; plus meme suicides

coiax opened the pull request:
:cl: coiax
rscadd: Airlock painters can be printed at your local autolathe
rscadd: Please do not attempt suicide with the airlock painter or the medical wrench.
/:cl:

coiax committed:
Adds airlock painter to autolathe designs

coiax committed:
Fixes spelling error in message

coiax committed:
Suicide memes: Airlock painter edition

Hope you have enough toner to commit suicide with, although I'm pretty
sure that all that dye is good for your lungs.

coiax committed:
Meme suicide: M E D I C A L W R E N C H

Pray to it, and it'll set you free of your earthly concerns.

ChangelingRain commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -50,6 +50,25 @@
+	W.add_fingerprint(user)
+	W.desc += " For some reason, it reminds you of [user.name]."
+
+	user.dust()
```
Probably bad, maybe have it drop items and teleport them to the morgue.

coiax replied:
So dropping the items where they stand is bad?

ChangelingRain replied:
Oh wait, it does that.

KorPhaeron commented:
Please put unrelated stuff in different PRs

ChangelingRain commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -50,6 +50,25 @@
+	sleep(20)
+
+	for (var/obj/O in user.contents)
+		O.loc = get_turf(src)
```
This is not actually the right way to do it; it should be
````
for(var/obj/item/W in user)
	user.unEquip(W, 1)
````
Otherwise it'll keep the UI layering and drop stuff like hardsuit helmets and shit.

KazeEspada replied:
NO, dont use 1! You will throw defibs and nozzles to the ground.

JJRcop commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -50,6 +50,25 @@
+
+	var/obj/item/weapon/wrench/medical/W = new /obj/item/weapon/wrench/medical(loc)
+	W.add_fingerprint(user)
+	W.desc += " For some reason, it reminds you of [user.name]."
```
So how long do you think it will take for an entire medbay to mass suicide using one wrench?

octareenroon91 replied:
This produces an additional medical wrench, right?

Mass suicide at O(log2(n)).

optimumtact commented on code/game/objects/items/weapons/airlock_painter.dm:
```diff
@@ -33,14 +33,52 @@
+		use(user)
+
+		// Wait two beats for the toner to take effect
+		sleep(20)
```
now you have to put a bunch of sanity checks after this to make sure the user still exists

optimumtact commented on code/game/objects/items/weapons/airlock_painter.dm:
```diff
@@ -33,14 +33,52 @@
+		playsound(V.loc, 'sound/effects/splat.ogg', 50, 1)
+
+		L.loc = get_turf(src)
+		sleep(10)
```
same again

PKPenguin321 replied:
why are the sleeps even needed

optimumtact commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -50,6 +50,25 @@
+	// TODO Make them glow with the power of the M E D I C A L W R E N C H
+
+	playsound(loc, 'sound/effects/pray.ogg', 50, 1, -1)
+	sleep(20)
```
sanity checks needed

Core0verload commented:
How about adding pipe painter's functions to airlock painter?

coiax committed:
Put in checks for medical wrench suicide

Better make sure user is still around and they haven't died during their
suicide.

coiax committed:
Vomit out my colorful lungs in front of me

Removed the sleeps, moved the lungs in front of you, added a case
for no lungs.

lordpidey commented on code/game/objects/items/weapons/airlock_painter.dm:
```diff
@@ -33,14 +33,62 @@
+
+		var/obj/item/organ/internal/lungs/L
+		L = user.getorganslot("lungs")
+		L.Remove(user)
```
Make sure L isn't null, in the event someone already lost their lungs, or this could runtime.

coiax committed:
Handled case where someone lost their lungs

We won't remove your lungs if you don't have any. But how did you
manage to lose them in this tiny window, you horrible person?

ChangelingRain commented on code/game/objects/items/weapons/airlock_painter.dm:
```diff
@@ -33,14 +33,67 @@
+		user.visible_message("<span class='suicide'>[user] vomits out their [L]!</span>")
+		playsound(user.loc, 'sound/effects/splat.ogg', 50, 1)
+
+		L.loc = turf
```
L.forceMove(turf)

ChangelingRain commented on code/game/objects/items/weapons/airlock_painter.dm:
```diff
@@ -33,14 +33,67 @@
+
+		// Find out if there is an open turf in front of us,
+		// and if not, pick the turf we are standing on.
+		var/turf
```
Should technically be something like var/turf/T even though you aren't using it for vars.

ChangelingRain commented on code/game/objects/items/weapons/airlock_painter.dm:
```diff
@@ -33,14 +33,67 @@
+		// and if not, pick the turf we are standing on.
+		var/turf
+
+		turf = get_step(get_turf(src), user.dir)
```
You can do this on line 58 with the var definition, no need to space it out.

ChangelingRain commented on code/game/objects/items/weapons/airlock_painter.dm:
```diff
@@ -33,14 +33,67 @@
+	if(user.getorganslot("lungs"))
+		has_lungs = TRUE
+
+	if(can_use(user) && has_lungs)
```
If has_lungs fails it'll do line 93 even if it has toner to use.

lordpidey replied:
consider changing this and the above few lines to
if(can_use(user) && user.getorganslot("lungs"))

optimumtact replied:
he's reusing the get has_lungs var at least once more, so I see what he's trying to do.

I see that he actually calls getorganslot lower down again on line 63.

I recommend
var/obj/item/organ/internal/lungs/L = user.getorganslot("lungs")

    if(can_use(user) && L)
        ...
    else if(can_use(user) && !L)
        ...
    else

that way you avoid the uncessary second call to getorganslot

coiax committed:
Use correct variable names and correct proc calls

coiax committed:
Removed unneeded seperation

coiax committed:
Only call get lungs once
